src/message_dao.py

```diff
@@ -12,27 +12,137 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, Optional
+import json
+import os
+import subprocess
+import zlib
+
+from functools import lru_cache
+from typing import Dict, List, Optional
+
+from dotenv import load_dotenv
+from google.cloud.sql.connector import Connector
 from message import Message
 
+load_dotenv()
 EPOCH_HASH = 'ae9e7be4a03765456fe38287533e6446e8bbc93c'
 
 class MessageDao(object):
+    def __init__(self) -> None:
+        self._initialize_connection()
+        self._initialize_tables()
+
+    def _initialize_connection(self) -> None:
+        connector = Connector()
+        self.connection = connector.connect(
+            os.environ.get("HOST"),
+            "pymysql",
+            user = os.environ.get("USER"),
+            password = os.environ.get("PASSWORD")
+        )
+    
+    def _initialize_tables(self) -> None:
+        db_name = os.environ.get("DB")
+        if not db_name:
+            raise Exception("Missing environment variable for name of database.")
+        with self.connection.cursor() as cursor:
+            cursor.execute("CREATE DATABASE IF NOT EXISTS " + db_name)
+            self.connection.select_db(db_name)
+            # Mapping from message id to message
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS Messages"
+                "(message_id VARCHAR(255) NOT NULL,"
+                "subject VARCHAR(255) NOT NULL,"
+                "from_ VARCHAR(255) NOT NULL,"
+                "in_reply_to VARCHAR(255),"
+                "content BLOB NOT NULL,"
+                "archive_hash VARCHAR(255),"
+                "change_id VARCHAR(255),"
+                "PRIMARY KEY (message_id))"
+            )
+            # Mapping from name of state attribute to 
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS States"
+                "(state_name VARCHAR(255) NOT NULL,"
+                "value VARCHAR(255) NOT NULL,"
+                "PRIMARY KEY (state_name))"
+            )
+        self.connection.commit()
+
+
+    def store(self, message: Message) -> None:
+        # Convert children into list of ids and store message
+        json_content = json.dumps(message.content)
+        comp_content = zlib.compress(json_content.encode())
```

Comment:
Is this necessary? How much space are we using? I would rather be able to look at the data in the database outside of the application.

Reply:
On average, we're saving 1.5k bytes of data by compressing it. However, the largest size after compression is 21k bytes vs 97k bytes prior to compression. It's not inherently clear which approach is better.

Reply:
Removed content and instead added a lore link for ease of access.